### ISSUE TITLE
AttributeError: '_Code' object has no attribute 'co_positions'

### ISSUE THREAD

**Issue by Radiergummi** [getsentry/sentry-python#1798]
### How do you use Sentry?

Sentry Saas (sentry.io)

### Version

1.12.1

### Steps to Reproduce

This is a bit hard to figure out. I installed the SDK as [documented](https://docs.sentry.io/platforms/python/) for a Django+Celery application, and have it running in Docker on Python 3.11.

What I see is that an unrelated exception is raised in our celery task (`celery/app/trace.py:265 in _log_error`), which passes it on to the Sentry SDK (`sentry_sdk/integrations/logging.py:96 in sentry_patched_callhandlers`), which seems to do something bad. Sorry, I'm not a Python dev :)

The error _is_ forwarded to Sentry correctly, however.

### Expected Result

Expected no logging errors to show up in our logs.

### Actual Result

```
[2022-12-20 20:40:18,704: WARNING/ForkPoolWorker-1] --- Logging error ---
[2022-12-20 20:40:18,705: WARNING/ForkPoolWorker-1] Traceback (most recent call last):
[2022-12-20 20:40:18,706: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/celery/app/trace.py", line 451, in trace_task
    R = retval = fun(*args, **kwargs)
                 ^^^^^^^^^^^^^^^^^^^^
[2022-12-20 20:40:18,706: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/sentry_sdk/integrations/celery.py", line 207, in _inner
    reraise(*exc_info)
[2022-12-20 20:40:18,706: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/sentry_sdk/_compat.py", line 57, in reraise
    raise value
[2022-12-20 20:40:18,706: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/sentry_sdk/integrations/celery.py", line 202, in _inner
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
[2022-12-20 20:40:18,706: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/celery/app/trace.py", line 734, in __protected_call__
    return self.run(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
[2022-12-20 20:40:18,706: WARNING/ForkPoolWorker-1]   File "/app/sia/tasks.py", line 116, in process_blocks
    process_blocks.delay(block_ids_unprocessed, retry_count=retry_count + 1)
[2022-12-20 20:40:18,706: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/celery/app/task.py", line 425, in delay
    return self.apply_async(args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2022-12-20 20:40:18,706: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/sentry_sdk/integrations/celery.py", line 127, in apply_async
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
[2022-12-20 20:40:18,706: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/celery/app/task.py", line 540, in apply_async
    check_arguments(*(args or ()), **(kwargs or {}))
[2022-12-20 20:40:18,706: WARNING/ForkPoolWorker-1] TypeError: process_blocks() missing 1 required positional argument: 'new_index_name'
[2022-12-20 20:40:18,706: WARNING/ForkPoolWorker-1]
During handling of the above exception, another exception occurred:
[2022-12-20 20:40:18,706: WARNING/ForkPoolWorker-1] Traceback (most recent call last):
[2022-12-20 20:40:18,706: WARNING/ForkPoolWorker-1]   File "/usr/local/lib/python3.11/logging/__init__.py", line 1110, in emit
    msg = self.format(record)
          ^^^^^^^^^^^^^^^^^^^
[2022-12-20 20:40:18,706: WARNING/ForkPoolWorker-1]   File "/usr/local/lib/python3.11/logging/__init__.py", line 953, in format
    return fmt.format(record)
           ^^^^^^^^^^^^^^^^^^
[2022-12-20 20:40:18,706: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/celery/utils/log.py", line 146, in format
    msg = super().format(record)
          ^^^^^^^^^^^^^^^^^^^^^^
[2022-12-20 20:40:18,706: WARNING/ForkPoolWorker-1]   File "/usr/local/lib/python3.11/logging/__init__.py", line 695, in format
    record.exc_text = self.formatException(record.exc_info)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2022-12-20 20:40:18,706: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/celery/utils/log.py", line 142, in formatException
    r = super().formatException(ei)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2022-12-20 20:40:18,707: WARNING/ForkPoolWorker-1]   File "/usr/local/lib/python3.11/logging/__init__.py", line 645, in formatException
    traceback.print_exception(ei[0], ei[1], tb, None, sio)
[2022-12-20 20:40:18,707: WARNING/ForkPoolWorker-1]   File "/usr/local/lib/python3.11/traceback.py", line 124, in print_exception
    te = TracebackException(type(value), value, tb, limit=limit, compact=True)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2022-12-20 20:40:18,707: WARNING/ForkPoolWorker-1]   File "/usr/local/lib/python3.11/traceback.py", line 690, in __init__
    self.stack = StackSummary._extract_from_extended_frame_gen(
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2022-12-20 20:40:18,707: WARNING/ForkPoolWorker-1]   File "/usr/local/lib/python3.11/traceback.py", line 416, in _extract_from_extended_frame_gen
    for f, (lineno, end_lineno, colno, end_colno) in frame_gen:
[2022-12-20 20:40:18,707: WARNING/ForkPoolWorker-1]   File "/usr/local/lib/python3.11/traceback.py", line 353, in _walk_tb_with_full_positions
    positions = _get_code_position(tb.tb_frame.f_code, tb.tb_lasti)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2022-12-20 20:40:18,707: WARNING/ForkPoolWorker-1]   File "/usr/local/lib/python3.11/traceback.py", line 366, in _get_code_position
    positions_gen = code.co_positions()
                    ^^^^^^^^^^^^^^^^^
[2022-12-20 20:40:18,707: WARNING/ForkPoolWorker-1] AttributeError: '_Code' object has no attribute 'co_positions'
[2022-12-20 20:40:18,707: WARNING/ForkPoolWorker-1] Call stack:
[2022-12-20 20:40:18,707: WARNING/ForkPoolWorker-1]   File "<frozen runpy>", line 198, in _run_module_as_main
[2022-12-20 20:40:18,707: WARNING/ForkPoolWorker-1]   File "<frozen runpy>", line 88, in _run_code
[2022-12-20 20:40:18,708: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/celery/__main__.py", line 19, in <module>
    main()
[2022-12-20 20:40:18,708: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/celery/__main__.py", line 15, in main
    sys.exit(_main())
[2022-12-20 20:40:18,708: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/celery/bin/celery.py", line 217, in main
    return celery(auto_envvar_prefix="CELERY")
[2022-12-20 20:40:18,708: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
[2022-12-20 20:40:18,708: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
[2022-12-20 20:40:18,708: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
[2022-12-20 20:40:18,708: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
[2022-12-20 20:40:18,708: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
[2022-12-20 20:40:18,708: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
[2022-12-20 20:40:18,708: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/celery/bin/base.py", line 134, in caller
    return f(ctx, *args, **kwargs)
[2022-12-20 20:40:18,708: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/celery/bin/worker.py", line 351, in worker
    worker.start()
[2022-12-20 20:40:18,708: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/celery/worker/worker.py", line 203, in start
    self.blueprint.start(self)
[2022-12-20 20:40:18,708: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/celery/bootsteps.py", line 116, in start
    step.start(parent)
[2022-12-20 20:40:18,708: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/celery/bootsteps.py", line 365, in start
    return self.obj.start()
[2022-12-20 20:40:18,708: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/celery/concurrency/base.py", line 129, in start
    self.on_start()
[2022-12-20 20:40:18,708: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/celery/concurrency/prefork.py", line 109, in on_start
    P = self._pool = Pool(processes=self.limit,
[2022-12-20 20:40:18,708: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/celery/concurrency/asynpool.py", line 463, in __init__
    super().__init__(processes, *args, **kwargs)
[2022-12-20 20:40:18,708: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/billiard/pool.py", line 1046, in __init__
    self._create_worker_process(i)
[2022-12-20 20:40:18,708: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/celery/concurrency/asynpool.py", line 480, in _create_worker_process
    return super()._create_worker_process(i)
[2022-12-20 20:40:18,708: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/billiard/pool.py", line 1158, in _create_worker_process
    w.start()
[2022-12-20 20:40:18,708: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/billiard/process.py", line 124, in start
    self._popen = self._Popen(self)
[2022-12-20 20:40:18,708: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/billiard/context.py", line 333, in _Popen
    return Popen(process_obj)
[2022-12-20 20:40:18,708: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/billiard/popen_fork.py", line 24, in __init__
    self._launch(process_obj)
[2022-12-20 20:40:18,708: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/billiard/popen_fork.py", line 79, in _launch
    code = process_obj._bootstrap()
[2022-12-20 20:40:18,708: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/billiard/process.py", line 327, in _bootstrap
    self.run()
[2022-12-20 20:40:18,708: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/billiard/process.py", line 114, in run
    self._target(*self._args, **self._kwargs)
[2022-12-20 20:40:18,708: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/billiard/pool.py", line 292, in __call__
    sys.exit(self.workloop(pid=pid))
[2022-12-20 20:40:18,708: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/sentry_sdk/integrations/celery.py", line 289, in sentry_workloop
    return old_workloop(*args, **kwargs)
[2022-12-20 20:40:18,709: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/billiard/pool.py", line 362, in workloop
    result = (True, prepare_result(fun(*args, **kwargs)))
[2022-12-20 20:40:18,709: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/celery/app/trace.py", line 649, in fast_trace_task
    R, I, T, Rstr = tasks[task].__trace__(
[2022-12-20 20:40:18,709: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/sentry_sdk/integrations/celery.py", line 184, in _inner
    return f(*args, **kwargs)
[2022-12-20 20:40:18,709: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/celery/app/trace.py", line 468, in trace_task
    I, R, state, retval = on_error(task_request, exc, uuid)
[2022-12-20 20:40:18,709: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/celery/app/trace.py", line 379, in on_error
    R = I.handle_error_state(
[2022-12-20 20:40:18,709: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/celery/app/trace.py", line 178, in handle_error_state
    return {
[2022-12-20 20:40:18,709: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/celery/app/trace.py", line 237, in handle_failure
    self._log_error(task, req, einfo)
[2022-12-20 20:40:18,709: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/celery/app/trace.py", line 265, in _log_error
    logger.log(policy.severity, policy.format.strip(), context,
[2022-12-20 20:40:18,709: WARNING/ForkPoolWorker-1]   File "/usr/local/lib/python3.11/logging/__init__.py", line 1559, in log
    self._log(level, msg, args, **kwargs)
[2022-12-20 20:40:18,709: WARNING/ForkPoolWorker-1]   File "/usr/local/lib/python3.11/logging/__init__.py", line 1634, in _log
    self.handle(record)
[2022-12-20 20:40:18,709: WARNING/ForkPoolWorker-1]   File "/usr/local/lib/python3.11/logging/__init__.py", line 1644, in handle
    self.callHandlers(record)
[2022-12-20 20:40:18,709: WARNING/ForkPoolWorker-1]   File "/home/pythonista/.local/lib/python3.11/site-packages/sentry_sdk/integrations/logging.py", line 96, in sentry_patched_callhandlers
    return old_callhandlers(self, record)
[2022-12-20 20:40:18,709: WARNING/ForkPoolWorker-1] Message: 'Task %(name)s[%(id)s] %(description)s: %(exc)s'
Arguments: {'hostname': 'celery@bc0647d7a482', 'id': 'afaa7036-3168-4db8-a52e-8e213c7f9730', 'name': 'sia.tasks.process_blocks', 'exc': 'TypeError("process_blocks() missing 1 required positional argument: \'new_index_name\'")', 'traceback': 'Traceback (most recent call last):\n  File "/home/pythonista/.local/lib/python3.11/site-packages/celery/app/trace.py", line 451, in trace_task\n    R = retval = fun(*args, **kwargs)\n                 ^^^^^^^^^^^^^^^^^^^^\n  File "/home/pythonista/.local/lib/python3.11/site-packages/sentry_sdk/integrations/celery.py", line 207, in _inner\n    reraise(*exc_info)\n  File "/home/pythonista/.local/lib/python3.11/site-packages/sentry_sdk/_compat.py", line 57, in reraise\n    raise value\n  File "/home/pythonista/.local/lib/python3.11/site-packages/sentry_sdk/integrations/celery.py", line 202, in _inner\n    return f(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^\n  File "/home/pythonista/.local/lib/python3.11/site-packages/celery/app/trace.py", line 734, in __protected_call__\n    return self.run(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File "/app/sia/tasks.py", line 116, in process_blocks\n    process_blocks.delay(block_ids_unprocessed, retry_count=retry_count + 1)\n  File "/home/pythonista/.local/lib/python3.11/site-packages/celery/app/task.py", line 425, in delay\n    return self.apply_async(args, kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File "/home/pythonista/.local/lib/python3.11/site-packages/sentry_sdk/integrations/celery.py", line 127, in apply_async\n    return f(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^\n  File "/home/pythonista/.local/lib/python3.11/site-packages/celery/app/task.py", line 540, in apply_async\n    check_arguments(*(args or ()), **(kwargs or {}))\nTypeError: process_blocks() missing 1 required positional argument: \'new_index_name\'\n', 'args': "[['000021961206', '000000261996', '000015904952', '000012476054', '000005124057', '000009799718', '000010777559', '000000047596', '000015503682', '000021608561'], 'index_2022_12_20_16_40_52']", 'kwargs': '{}', 'description': 'raised unexpected', 'internal': False}
```

**Comment by sl0thentr0py** [comment]
@Radiergummi thx for the report.
What I believe is happening is:
* your code throws the exception `TypeError: process_blocks() missing 1 required positional argument: 'new_index_name'` which sentry picks up, sends and then re-raises
* celery then tries to log this exception and it's logger is throwing because [python3.11 added `co_positions`](https://peps.python.org/pep-0657/#specification) and something between celery<->3.11 is off

**Comment by sl0thentr0py** [comment]
can you also give me your celery version?

**Comment by sl0thentr0py** [comment]
also we haven't officially tested 3.11 yet so we'll get to that as well

**Comment by Radiergummi** [comment]
Im still on vacation until Monday, will get back to this comment next week!

**Comment by ben-steplab** [comment]
Hey - I'm also getting this bug using Python 3.11 - any updates are welcome. Using `celery==5.2.7`, `sentry-sdk==1.13.0`.

Thanks

**Comment by Radiergummi** [comment]
> can you also give me your celery version?

@sl0thentr0py Sorry! Had this on my todo for a while. Like @ben-steplab, we're on `celery==5.2.7`.

**Comment by marns93** [comment]
@sl0thentr0py @antonpirker We are also facing this issue with Python 3.11 using `celery==5.2.7`, `sentry-sdk==1.15.0`. Any updates on this?

**Comment by lordoffreaks** [comment]
Same here using `sentry-sdk==1.17.0` and `celery==5.2.7` ... any updates?

**Comment by antonpirker** [comment]
Hey, no updates right now. We have a lot on our plates right now, but we have it on our radar.

**Comment by sentrivana** [comment]
Hi all, thanks for reporting this. I did some research and it seems like this is due to one of `celery`'s dependencies not fully supporting py3.11, see https://github.com/celery/billiard/issues/377. The issue will be fixed in `celery>=5.3` which is not out yet, but at the time of writing there's a beta version `5.3.0b2` which gets rid of the issue.

We can probably just wait for 5.3 to come out.
